src/kagglehub/handle.py

```diff
@@ -34,6 +34,11 @@ class ModelHandle(ResourceHandle):
     def is_versioned(self) -> bool:
         return self.version is not None and self.version > 0
 
+    def with_version(self, version: int):  # noqa: ANN201
```

Comment:
instead of using `noqa` to ignore the warning, can you simply define the return type?

```
def with_version(self, version: int) -> ModelHandle:
```

Reply:
Unfortunately not, you can't refer to your defining class inside yourself :\ there is `typing.Self` for this purpose, but it didn't work with Python 3.9 I think.  There was another workaround that was also a bit messy so I thought to just denylist it... can take another pass though.